source/connection-monitoring-and-pooling/tests/README.md

```diff
@@ -21,6 +21,51 @@ The following tests have not yet been automated, but MUST still be tested:
 5. When a check out attempt fails because connection set up throws an error, assert that a ConnectionCheckOutFailedEvent
     with reason="connectionError" is emitted.
 
+### Pending Response: Connection Aliveness Tests
+
+If a connection with a pending response is idle for > 3 seconds, then drivers are expected to perform an aliveness check
+by attempting a non-blocking read of 1 byte from the inbound TCP buffer. The following two cases test both a successful
+read and a failed one.
+
+#### Connection Aliveness Check Fails
+
+1. Initialize a mock TCP listener to simulate the server-side behavior. The listener should write at least 5 bytes to
+    the connection to prevent size-related errors (e.g. `0x01, 0x02, 0x03, 0x04, 0x05, 0xFF`). The response should be
+    delayed by 2x the size of the socket timeout.
+2. Implement a monitoring mechanism to capture the `ConnectionPendingResponseStarted` and
+    `ConnectionPendingResponseFailed` events.
+3. Instantiate a connection pool using the mock listener’s address, ensuring readiness without error. Attach the event
+    monitor to observe the connection’s state.
+4. Check out a connection from the pool and initiate a read operation with an appropriate socket timeout (e.g, 10ms)
+    that will trigger a timeout due to the artificial delay of 2x the socket timeout (step 1). Ensure that the read
+    operation returns a timeout error.
+5. Check the connection back into the pool and sleep for 3 seconds so that the pending response state timestamp exceeds
+    the pending response timeout, forcing an aliveness check.
+6. Check the connection out. The aliveness check should fail since no additional bytes were added after the delay in
+    step 1.
+7. Verify that one event for each `ConnectionPendingResponseStarted` and `ConnectionPendingResponseFailed` was emitted.
+    Also verify that the fields were correctly set for each event.
+
+#### Connection Aliveness Check Succeeds
```

Comment:
Can we add a test that demonstrates that multiple aliveness checks might be required to fully read the response from the socket?  I'm imagining the mock server emitting chunks of data every second for longer than 6s (2 * the static timeout).  Each checkout should fail, but we'll continue to read

Reply:
It’s unclear to me what the goal of this would be. Are you just wanting to make sure the driver does not pre-maturely close the connection if the aliveness check succeeds? We could use event monitoring for that instead.

Reply:
Yes - I want a test somewhere that demonstrates that drivers might need to span multiple checkout + pending responses, and that that process can take longer than the static timeout, before reading the full response from the connection.

Reply:
I suggest we defer this to the unified spec test and not further complicate the prose aliveness checks. Validating that a successful aliveness check does not result in a closed connection would imply that a subsequent check out would either perform an aliveness check or continue attempting to drain the buffer. LMK what you think.